### PR TITLE
DROOLS-3229 DMN compilation of DT with empty inputValues, outputValues,

### PR DESCRIPTION
**Pull request by tarilabs**
... defaultOutputEntry NPEs

/cc @etirelli @baldimir @jiripetrlik

### CODE REVIEW

**Review by baldimir — APPROVED** [review]

File: kie-dmn/kie-dmn-core/src/test/java/org/kie/dmn/core/DMNRuntimeTest.java [hunk 1]
```diff
@@ -2362,5 +2362,29 @@ public void testAssignNullToAllowedValuesExplicitingNull() {
...
+        final DMNRuntime runtime = DMNRuntimeUtil.createRuntime("EmptyinputValuesoutputValuesdefaultOutputEntry.dmn", this.getClass());
+        final DMNModel dmnModel = runtime.getModel("http://www.trisotech.com/dmn/definitions/_31f2dde9-b27c-41f8-97b4-5c8dd728942e", "Drawing 1");
+        assertThat(dmnModel, notNullValue());
+        assertThat(DMNRuntimeUtil.formatMessages(dmnModel.getMessages()), dmnModel.hasErrors(), is(false));
+
+        checkEmptyinputValuesoutputValuesdefaultOutputEntry(runtime, dmnModel, 47, "positive");
+        checkEmptyinputValuesoutputValuesdefaultOutputEntry(runtime, dmnModel, -1, "negative");
+    }
+
+    private static void checkEmptyinputValuesoutputValuesdefaultOutputEntry(final DMNRuntime runtime, final DMNModel dmnModel, int i, String string) {
```
Comment (jomarko) [inline kie-dmn/kie-dmn-core/src/test/java/org/kie/dmn/core/DMNRuntimeTest.java]: Hi @tarilabs looks good to me, just minor comments to make reading more easy in future, when the issue won't be so fresh in our minds.
- Could you please rename `i` to `testedInput`, `testedNumber` or something like that?
- Could you please rename `string` to `expectedResult` or something similar?
- Not sure if Camel Case for this method shouldn't be `checkEmptyInputValuesOutputValuesDefaultOutputEntry`

Comment (tarilabs) [reply kie-dmn/kie-dmn-core/src/test/java/org/kie/dmn/core/DMNRuntimeTest.java]: It's just the name of the file

Comment (jomarko) [reply kie-dmn/kie-dmn-core/src/test/java/org/kie/dmn/core/DMNRuntimeTest.java]: @tarilabs thanks for updates, one more question, just to learn things, tho whole decision table, returns "string", right? I am not getting why we can use different type ref for default output, boolean in this case, see

```
<semantic:defaultOutputEntry id="_6C115712-ED1D-4A20-97B1-709BC8F3E50C" typeRef="boolean">
  <semantic:text></semantic:text>
</semantic:defaultOutputEntry>
```

Comment (tarilabs) [reply kie-dmn/kie-dmn-core/src/test/java/org/kie/dmn/core/DMNRuntimeTest.java]: Now this is an awesome feedback, thanks!

**Review by jomarko — APPROVED** [review]
No more comments from my site, thanks Matteo.

**Pull request merged**